sdk/storage/Azure.Storage.Common/tests/Shared/TestHelper.cs

```diff
@@ -61,7 +61,24 @@ public static void AssertExpectedException<T>(Action action, T expectedException
 
         public static void AssertExpectedException<T>(Action action, Func<T, bool> predicate = null)
             where T : Exception
-            => AssertExpectedException(action, default, GetDefaultExceptionAssertion<T>((_, a) => predicate(a)));
+        {
```

Comment:
Should you flip it around and make the other version call this then with a `predicate: _ => true`?

Reply:
Which other version do you mean? The one that takes `expectedException`?

The problem here was that existing implementation of this overload ended up in [this line](https://github.com/Azure/azure-sdk-for-net/blob/2b594335737a65cb8f36543ef817deaba5c43f4b/sdk/storage/Azure.Storage.Common/tests/Shared/TestHelper.cs#L46-L49) attempting to assert exception message, which was deemed to always fail... This overload of `AssertExpectedException` had zero callers because of that problem.